src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -362,6 +364,7 @@ private void OnBatchScopeDisposed()
         {
             lock (_gate)
             {
+                Action fireEvents = null;
```

Comment:
if you give this a non-null value (like `() => {}`), then the code below becomes much cleaner.